quaiclient/ethclient/ethclient.go

```diff
@@ -138,12 +138,18 @@ func (ec *Client) getBlock(ctx context.Context, method string, args ...interface
 	if head.EtxHash() != types.EmptyRootHash && len(body.ExtTransactions) == 0 {
 		return nil, fmt.Errorf("server returned empty external transaction list but block header indicates transactions")
 	}
-	if head.ManifestHash() == types.EmptyRootHash && len(body.SubManifest) > 0 {
-		return nil, fmt.Errorf("server returned non-empty subordinate manifest but block header indicates no transactions")
-	}
-	if head.ManifestHash() != types.EmptyRootHash && len(body.SubManifest) == 0 {
-		return nil, fmt.Errorf("server returned empty subordinate manifest but block header indicates transactions")
-	}
+
+	// TODO: Fix the checking here on manifest hashes in RPC
```

Comment:
The submanifest should be empty because there are no ETXs. Is it not empty?

Reply:
failed on the second case "header indicates transactions" in the prime context.